"""Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT."""

from sdk import SDK, SERVERS, SERVER_DEFAULT_SERVER, SERVER_BROKEN_SERVER
from .common_helpers import record_test

def test_select_global_server_valid():
    record_test('servers-select-global-server-valid')

    url = SERVERS[SERVER_DEFAULT_SERVER]
    assert url == "http://localhost:35123"

    s = SDK(server_url=url)
    assert s is not None

    res = s.servers.select_global_server()

    assert res is not None


def test_select_global_server_broken():
    record_test('servers-select-global-server-broken')

    url = SERVERS[SERVER_BROKEN_SERVER]
    assert url == "http://broken"

    s = SDK(server_url=url)
    assert s is not None

    error = None

    try:
        s.servers.select_global_server()
    except Exception as err:
        error = err

    assert error is not None


def test_select_global_server_by_name_default():
    record_test('servers-select-global-server-by-name-default')

    s = SDK()
    assert s is not None

    res = s.servers.select_global_server()

    assert res is not None


def test_select_global_server_by_name_valid():
    record_test('servers-select-global-server-by-name-valid')

    assert SERVER_DEFAULT_SERVER == 'default-server'
    s = SDK(server=SERVER_DEFAULT_SERVER)
    assert s is not None

    res = s.servers.select_global_server()

    assert res is not None


def test_select_global_server_by_name_invalid():
    record_test('servers-select-global-server-by-name-invalid')

    error = None
    try:
        s = SDK(server='unknown')
    except Exception as err:
        error = err

    assert error is not None
    assert str(error) == "Invalid server \"unknown\""

def test_select_global_server_by_name_broken():
    record_test('servers-select-global-server-by-name-broken')

    assert SERVER_BROKEN_SERVER == 'BrokenServer'
    s = SDK(server=SERVER_BROKEN_SERVER)
    assert s is not None

    error = None

    try:
        s.servers.select_global_server()
    except Exception as err:
        error = err

    assert error is not None
